### ISSUE TITLE
Support checking for Multinomial is unnecessarily multidimensional

### ISSUE THREAD

**Issue by aravikumar0** [pyro-ppl/numpyro#1559]
A multidimensional array is constructed when checking the support for a multinomial with total count broadcasted across multidimensional probs or logits.
```python
mixing_dist = dist.CategoricalProbs(
    probs=np.array([0.4, 0.1, 0.1, 0.1, 0.1])
)
component_dist = dist.MultinomialProbs(
    total_count=jnp.array([100, 200, 300])[:, None],
    probs=jnp.array(
        [
            [0.4, 0.5, 0.1],
            [0.9, 0.05, 0.05],
            [0.01, 0.8, 0.19],
            [0.2, 0.4, 0.4],
            [0.1, 0.1, 0.8]
        ]
    )
)
sample_dist = dist.MixtureSameFamily(
    mixing_dist,
    component_dist
)
Y = sample_dist.sample(jrandom.PRNGKey(0))
sample_dist.support(Y)
```
returns
```python
Array([[ True, False, False],
       [False,  True, False],
       [False, False,  True]], dtype=bool)
```
 We only need to check the diagonals of this multidimensional array for valid support. In big models, this sometimes results in an OOM error.

Subclassing the MultinomialLogits / MultinomialProbs distribution and modifying the `support` method fixes this.

```python
@dist.constraints.dependent_property(is_discrete=True, event_dim=1)
def support(self):
    return dist.constraints.multinomial(self.total_count.squeeze())
```

Versions:
`numpyro == 0.11.0`
`jax == 0.4.6`

**Comment by fehiepsi** [comment]
> We only need to check the diagonals of this multidimensional array for valid support.

Hi @aravikumar0, if you don't want to validate some distribution, you can set `validate_args=False`. Like `dist.MultinomialProbs(..., validate_args=False)`.

**Comment by aravikumar0** [comment]
Thank you!
